litr/src/main/java/com/linkedin/android/litr/frameextract/behaviors/MediaMetadataExtractBehavior.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 LinkedIn Corporation
+ * All Rights Reserved.
+ *
+ * Licensed under the BSD 2-Clause License (the "License").  See License in the project root for
+ * license information.
+ */
+package com.linkedin.android.litr.frameextract.behaviors
+
+import android.content.Context
+import android.media.MediaMetadataRetriever
+import android.net.Uri
+import com.linkedin.android.litr.ExperimentalFrameExtractorApi
+import com.linkedin.android.litr.frameextract.FrameExtractMode
+import com.linkedin.android.litr.frameextract.FrameExtractParameters
+
+@ExperimentalFrameExtractorApi
```

Comment:
Do we still need this here? Does this get inherited from parent interface?

Reply:
Yeah I believe we still need this -- basically this is propagating the opt-in requirement to whoever is using this concrete behavior. If I leave this out, we'll get warnings in this class that it's using an experimental API. So we have two choices: opt in, or propagate. If we opt in here, then callers won't know this class is using experimental APIs. So instead we propagate the requirement. https://kotlinlang.org/docs/opt-in-requirements.html#propagating-opt-in